docs: clarify that mtime and mode are optional (#3302)
ipfs/js-ipfs: docs/core-api/FILES.md

| -------- | -------- |
| `Promise<UnixFSEntry>` | A object describing the added data |

> Each yielded object is of the form (`mtime` and `mode` are optional, so may be missing):

Reviewer: `mode` has a default value so shouldn't be missing.

https://github.com/ipfs/specs/blob/master/UNIXFS.md#metadata

Author: @achingbrain Ack, I was unaware of implicit `mode` default. 
This complicates things a bit.. would this work?

```suggestion
Each yielded object is of the form (with optional `mtime` and custom/implicit `mode` [metadata](https://github.com/ipfs/specs/blob/master/UNIXFS.md#metadata)):
```

If not, we could just leave this line as it was before. Let me know.

Reviewer 2: Perhaps reverting this and just adding a comment in the JS object below would be helpful? Additionally, keeping the optional `?` indicator on the key is a nice way to represent the potential missing response value.

```js
{
  path: '/tmp/myfile.txt',
  cid: CID('QmHash'),
  mode: Number, // implicit if not provided
  mtime?: { secs: Number, nsecs: Number }, // may be missing
  size: 123
}
```
